tests/integration/docidcache/test_crud_cache.py

```diff
@@ -0,0 +1,226 @@
+import os
+
+import numpy as np
+import pytest
+
+from jina import Flow, Document
+from jina.executors.indexers import BaseIndexer
+from jina.executors.indexers.cache import DocIDCache
+from jina.executors.indexers.keyvalue import BinaryPbIndexer
+from tests.integration.incremental_indexing import random_workspace
+from pathlib import Path
+
+cur_dir = Path(os.path.dirname(os.path.abspath(__file__)))
+
+# don't remove this line, otherwise auto-code-format will remove `random_workspace`
+print(random_workspace)
+
+
+def config(field, tmp_workspace):
+    os.environ['JINA_CACHE_FIELD'] = field
+    os.environ['JINA_TEST_CACHE_CRUD_WORKSPACE'] = str(tmp_workspace)
+
+
+def get_flow(indexers, shards):
+    vector_pod = dict(
+        uses=cur_dir / 'yml' / 'cp_cache_pb.yml',
+        name='inc_vec'
+    )
+    pb_pod = dict(
+        uses=cur_dir / 'yml' / 'cp_cache_vector.yml',
+        name='inc_doc'
+    )
+
+    if shards == 3:
```

Comment:
You can have this config in the `pb_pod` and have `shards` parametrized. This construction is too noisy.

`separated_workspace`, `polling` and `uses_after` will not affect if shards=1

Reply:
I do prefer to have the `ymls` parametrized with `environment variables` (or moustache ones) so that just reading the yaml files, it is clear what is being tested.

Reply:
I would still need to keep it as code for the parallel (`needs`) part, right?

Reply:
needs and parallel are independent things, but u can also paramrtrize a list with env variable. I think is more readable to have flows in yaml files